src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -398,5 +408,17 @@ public static void ValidateCredentials(string username, string password, string
                 throw new CommandException(Strings.SourcesCommandCredentialsRequiredWithAuthTypes);
             }
         }
+
+        public static int ParseProtocolVersion(string protocolVersion)
+        {
+            var currentlySupportedProtocolVersions = new List<int> { 2, 3, 4 };
```

Comment:
I'm not aware of protocol version 4. Is it documented somewhere?

Reply:
I've found is referenced here <https://learn.microsoft.com/en-us/nuget/api/nuget-protocols#nuget-protocol-version-410> trying to find some list of valid versions and was really confused. Should I leave it or remove it?
I know that other places in code reference 3 as the max version.

Reply:
@joelverhagen
Which versions are available? If I understand correctly only v2, v3 are available.

Reply:
The naming here is pretty confusing. V2 (OData), V3 (Service index, JSON) are the protocol versions most people talk about. The `X-NuGet-Protocol-Version` header referenced here is a header that allows feeds (mainly NuGet.org) to reject older client implementations specifically on the push endpoint. The value could have been anything. `4.1.0` was picked because the original header name was "client version" but then we changed that a bit later on. So `4.1.0` is actually the first NuGet _client_ version that is compatible with the latest breaking change added to package push on NuGet.org.

I don't want to write a book here, so I'll just say that `2, 3` is correct. `2, 3, 4` is not correct.

See https://learn.microsoft.com/en-us/nuget/api/overview#versioning for more (general, not NuGet.org push specific) protocol versions info.

Reply:
Thanks for clarification.
I've pushed update with only `2, 3` as correct versions.